src/execution/operator/join/physical_comparison_join.cpp

```diff
@@ -26,6 +26,18 @@ PhysicalComparisonJoin::PhysicalComparisonJoin(LogicalOperator &op, PhysicalOper
 
 string PhysicalComparisonJoin::ParamsToString() const {
 	string extra_info = EnumUtil::ToString(join_type) + "\n";
+	switch (join_type) {
+	case JoinType::LEFT_SEMI:
+	case JoinType::LEFT_ANTI:
+		extra_info = "LEFT " + extra_info;
+		break;
+	case JoinType::RIGHT_SEMI:
+	case JoinType::RIGHT_ANTI:
+		extra_info = "RIGHT " + extra_info;
+		break;
+	default:
+		break;
+	}
```

Comment:
If we use `scripts/generate_enum_util.py` (see my other comment), we don't have to add the special case here.

Reply:
I did use that, the problem was that when the validator tries to convert a plan back to SQL I don't want the join type to return "LEFT SEMI" or "RIGHT SEMI" because we don't have parser support for that yet.
But when we print logical plans and such, I would like to know if the semi is a right semi or left semi

Reply:
The point remains: we can't manually edit `enum_util.cpp`. The next time someone adds a field to an enum, they will want to run `generate_enum_util.py`, which will overwrite your changes. They would then have to apply this patch every time manually.

I don't think we want to add parser support for that either, so we would need to special-case the conversion back to SQL instead of editing this file. Maybe we can have a look tomorrow at how that should be done